internal/services/expiry-checker.go

```diff
@@ -0,0 +1,71 @@
+package services
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/babylonlabs-io/babylon-staking-indexer/internal/types"
+	"github.com/babylonlabs-io/babylon-staking-indexer/internal/utils/poller"
+	"github.com/rs/zerolog/log"
+)
+
+func (s *Service) StartExpiryChecker(ctx context.Context) {
+	expiryCheckerPoller := poller.NewPoller(
+		s.cfg.Poller.ExpiryCheckerPollingInterval,
+		s.checkExpiry,
+	)
+	go expiryCheckerPoller.Start(ctx)
+}
+
+func (s *Service) checkExpiry(ctx context.Context) *types.Error {
+	btcTip, err := s.btc.GetBlockCount()
+	if err != nil {
+		return types.NewInternalServiceError(
+			fmt.Errorf("failed to get BTC tip height: %w", err),
+		)
+	}
+
+	expiredDelegations, err := s.db.FindExpiredDelegations(ctx, uint64(btcTip))
+	if err != nil {
+		return types.NewInternalServiceError(
+			fmt.Errorf("failed to find expired delegations: %w", err),
+		)
+	}
+
+	for _, delegation := range expiredDelegations {
+		delegation, err := s.db.GetBTCDelegationByStakingTxHash(ctx, delegation.StakingTxHashHex)
+		if err != nil {
+			return types.NewError(
+				http.StatusInternalServerError,
+				types.InternalServiceError,
+				fmt.Errorf("failed to get BTC delegation by staking tx hash: %w", err),
+			)
+		}
+
+		// previous state should be unbonding
+		if delegation.State != types.StateUnbonding {
+			return types.NewError(
+				http.StatusInternalServerError,
+				types.InternalServiceError,
+				fmt.Errorf("BTC delegation is not in unbonding state"),
+			)
+		}
+
+		if err := s.db.UpdateBTCDelegationState(ctx, delegation.StakingTxHashHex, types.StateWithdrawable); err != nil {
+			log.Error().Err(err).Msg("Error updating BTC delegation state to withdrawable")
+			return types.NewInternalServiceError(
+				fmt.Errorf("failed to update BTC delegation state to withdrawable: %w", err),
+			)
+		}
+
+		if err := s.db.DeleteExpiredDelegation(ctx, delegation.StakingTxHashHex); err != nil {
```

Comment:
just curious, after delegation, how could a user know the current delegation state?